diff --git a/dist/NodeChildProcessSpawner.js b/dist/NodeChildProcessSpawner.js
index 384b19c2265cb753509f7c988cb6b450042d975e..e009c950a4d9d0c7366b44c84d5a859c33cef65d 100644
--- a/dist/NodeChildProcessSpawner.js
+++ b/dist/NodeChildProcessSpawner.js
@@ -336,13 +336,15 @@ const make = /*#__PURE__*/Effect.gen(function* () {
           let isReferenced = true;
           const cwd = yield* resolveWorkingDirectory(cmd.options);
           const env = resolveEnvironment(cmd.options);
+          const commandOptions = cmd.options;
           const stdio = buildStdioArray(stdinConfig, stdoutConfig, stderrConfig, resolvedAdditionalFds);
           const [childProcess, exitSignal] = yield* Effect.acquireRelease(spawn(cmd, {
             cwd,
             env,
             stdio,
             detached: cmd.options.detached ?? process.platform !== "win32",
-            shell: cmd.options.shell
+            shell: cmd.options.shell,
+            windowsHide: commandOptions.windowsHide
           }), Effect.fnUntraced(function* ([childProcess, exitSignal]) {
             const exited = yield* Deferred.isDone(exitSignal);
             const killWithTimeout = withTimeout(childProcess, cmd, cmd.options);
diff --git a/src/NodeChildProcessSpawner.ts b/src/NodeChildProcessSpawner.ts
index ec7da1d3444463ee4883411e1cc2a2ce14ef9644..a88217f388eb419e8b9ca26168602dd366713363 100644
--- a/src/NodeChildProcessSpawner.ts
+++ b/src/NodeChildProcessSpawner.ts
@@ -468,6 +468,9 @@ const make = Effect.gen(function*() {
 
         const cwd = yield* resolveWorkingDirectory(cmd.options)
         const env = resolveEnvironment(cmd.options)
+        const commandOptions = cmd.options as ChildProcess.CommandOptions & {
+          readonly windowsHide?: boolean | undefined
+        }
         const stdio = buildStdioArray(stdinConfig, stdoutConfig, stderrConfig, resolvedAdditionalFds)
 
         const [childProcess, exitSignal] = yield* Effect.acquireRelease(
@@ -476,7 +479,8 @@ const make = Effect.gen(function*() {
             env,
             stdio,
             detached: cmd.options.detached ?? process.platform !== "win32",
-            shell: cmd.options.shell
+            shell: cmd.options.shell,
+            windowsHide: commandOptions.windowsHide
           }),
           Effect.fnUntraced(function*([childProcess, exitSignal]) {
             const exited = yield* Deferred.isDone(exitSignal)
